File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int rem,sum=0;
        while(x){
            rem = x%10;
            if((sum > INT_MAX/10) || (sum < INT_MIN/10)){
                return 0;
            }
            sum = (sum*10) + rem;
            x/=10;

        }
        
        return sum;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        ans=0
        sign=0
        if(x<0):
            sign=-1
        else:
            sign=1
        x=abs(x)
        while(x):
            rem=x%10
            ans=ans*10+rem
            x=x//10
        ans=ans*sign
        if(ans<-2**31 or ans>2**31-1):
            return 0
        return ans
        """
        :type x: int
        :rtype: int
        """
        